test/test_filter_map.cpp

```diff
@@ -0,0 +1,164 @@
+
+// Copyright (c) 2024 Tristan Brindle (tcbrindle at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include "catch.hpp"
+
+#include <array>
+#include <utility>
```

Comment:
Looking at the [CI failure](https://github.com/tcbrindle/flux/actions/runs/9971483316/job/27554724376#step:6:308), it seems like you need to `#include <optional>` here